packages/worker/src/workflow/vm.ts

```diff
@@ -19,13 +19,51 @@ interface ActivationContext {
 // Not present in @types/node for some reason
 const { promiseHooks } = v8 as any;
 
+/**
+ * Variant of {@link cutoffStackTrace} that works with FileLocation, keep this in sync with the original implementation
```

Comment:
There's more and more source map and stack traces manipulations sparkled all over the place... It will be difficult to keep everything in sync and it shadows `vm.ts`'s actual responsibilities. Not high priority, but it be pertinent to at least gather some part of this logic to some central place.

Reply:
Agree 👍

Reply:
I didn't want to put this in common since I didn't want to export these functions to users so just embedded it here.
Tests should protect us from desync.